__config__/webpack/webpack.config.js

```diff
@@ -1,76 +1,52 @@
 // @flow
-/* eslint-disable global-require,no-console */
+/* eslint-disable global-require,no-console,import/no-dynamic-require */
 const path = require('path');
-const webpack = require('webpack');
-const merge = require('webpack-merge');
 
-const production = process.env.NODE_ENV === 'production';
+// https://github.com/survivejs/webpack-merge#smart-merging
+const { smart: merge } = require('webpack-merge');
 
-const babelLoader = env => ({
-    test: /\.(js|jsx)$/,
-    exclude: /node_modules/,
-    use: {
-        loader: 'babel-loader',
-        options: { envName: env },
-    },
-});
+const dist = path.resolve(__dirname, '../../dist');
 
-const baseConfig = {
+const config = platform => ({
+    name: platform,
+    entry: { app: `./src/${platform}.js` },
     output: {
-        path: path.resolve(__dirname, '../../dist'),
+        path: dist,
+        filename: `[name].${platform}.js`,
+        chunkFilename: `[name].${platform}.js`,
+        publicPath: '/assets/javascript/',
     },
     stats: 'errors-only',
-    plugins: [
-        new webpack.DefinePlugin({
-            'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV),
-        }),
-        new webpack.NamedModulesPlugin(),
-    ],
-};
+    module: {
+        rules: [
+            {
+                test: /\.(js|jsx)$/,
+                exclude: /node_modules/,
+                use: {
+                    loader: 'babel-loader',
+                    options: { envName: `app:${platform}` },
+                },
+            },
+        ],
+    },
+});
 
-const envConfig = production
-    ? require('./webpack.config.prod')({
-          dist: baseConfig.output.path,
-          bundleName: 'app.browser.js',
-      })
-    : require('./webpack.config.dev');
+const envConfig = require(`./webpack.config.${process.env.NODE_ENV}`)({
+    dist,
+    bundleName: 'app.browser.js',
```

Comment:
Do we need to pass this through to the config? I think it's only used in one place. Can it be hardcoded?

Reply:
good point